### PR TITLE
[IMP] account, stock_landed_costs, web:  Invoice line display

### PR DESCRIPTION
**Pull request by malb-odoo**
This commit will do multiple things:
- Put a min-width and max width on the product label selected widget
- Change the label of analytic distribution
- Move landed costs before quantity and put it in optional hide
- Fix the accepted props type of listViewWidth to accept list of two elements

task: 4170328

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/account_move_views.xml [hunk 1]
```diff
@@ -1075,6 +1075,7 @@
                                                domain="[('deprecated', '=', False), ('account_type', 'not in', ('asset_receivable', 'liability_payable', 'off_balance')), ('company_id', 'parent_of', parent.company_id)]"
                                                required="display_type not in ('line_note', 'line_section')"/>
                                         <field name="analytic_distribution" widget="analytic_distribution"
+                                               string="Analytic"
```
Comment (malb-odoo) [inline addons/account/views/account_move_views.xml:1078]: need to add the pot

File: addons/web/static/src/views/fields/field.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ fieldRegistry.addValidation({
     },
     useSubView: { type: Boolean, optional: true },
     label: { type: [String, { value: false }], optional: true },
-    listViewWidth: { type: [Number, [Number, Number], Function], optional: true },
+    listViewWidth: { type: [Number, { type: Array, validate: (array) => array.length === 2 }, Function], optional: true },
```
Comment (kebeclibre) [inline addons/web/static/src/views/fields/field.js:70]: Why ?

Comment (malb-odoo) [reply addons/web/static/src/views/fields/field.js:70]: In debug mode i have an error that the type is not accepted

Comment (malb-odoo) [reply addons/web/static/src/views/fields/field.js:70]: I suppose it does not recognize the array with the [Number, Number] ?

Comment (aab-odoo) [inline addons/web/static/src/views/fields/field.js:70]: Yep, that's my fault, `[Number, Number]` means number or number, not pair of numbers.
```suggestion
    listViewWidth: { type: [Number, { type: Array, elements: Number, validate: (array) => array.length === 2 }, Function], optional: true },
```

**Comment by chklop** [comment]
We should merge this ASAP for the OXP. It drastically improves the UX for Invoice/SO/PO edition.

**Review by jbw-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**